### ISSUE TITLE
when i run aeron-mdc use docker-compose ,i got a problem

### ISSUE THREAD

**Issue by supergaga** [eleventy7/aeron-cookbook-code#102]
i run   `docker compose up` ,and i got a problem

here is the error log

```
➜  aeron-mdc git:(main) docker compose up
[+] Running 3/3
 ⠿ Container aeron-mdc_mdc-subscriber-1_1  Recreated                                                                                                                            0.2s
 ⠿ Container aeron-mdc_mdc-publisher_1     Recreated                                                                                                                            0.2s
 ⠿ Container aeron-mdc_mdc-subscriber-2_1  Recreated                                                                                                                            0.2s
Attaching to mdc-publisher_1, mdc-subscriber-1_1, mdc-subscriber-2_1
mdc-subscriber-1_1  | 12:38:00.007 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent - launching media driver
mdc-subscriber-2_1  | 12:38:00.132 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent - launching media driver
mdc-subscriber-1_1  | 12:38:00.152 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent - connecting aeron; media driver directory /dev/shm/aeron-root
mdc-subscriber-1_1  | 12:38:00.164 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent - adding the subscription
mdc-subscriber-1_1  | Exception in thread "main" java.util.NoSuchElementException
mdc-subscriber-1_1  | 	at java.base/java.net.NetworkInterface$1.nextElement(NetworkInterface.java:415)
mdc-subscriber-1_1  | 	at com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent.localHost(MultiDestinationSubscriberAgent.java:89)
mdc-subscriber-1_1  | 	at com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent.<init>(MultiDestinationSubscriberAgent.java:48)
mdc-subscriber-1_1  | 	at com.aeroncookbook.aeron.mdc.MultiDestinationSubscriber.main(MultiDestinationSubscriber.java:28)
mdc-publisher_1     | 12:38:00.273 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationPublisherAgent - launching MediaDriver
mdc-subscriber-2_1  | 12:38:00.290 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent - connecting aeron; media driver directory /dev/shm/aeron-root
mdc-subscriber-2_1  | 12:38:00.303 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent - adding the subscription
mdc-subscriber-2_1  | Exception in thread "main" java.util.NoSuchElementException
mdc-subscriber-2_1  | 	at java.base/java.net.NetworkInterface$1.nextElement(NetworkInterface.java:415)
mdc-subscriber-2_1  | 	at com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent.localHost(MultiDestinationSubscriberAgent.java:89)
mdc-subscriber-2_1  | 	at com.aeroncookbook.aeron.mdc.MultiDestinationSubscriberAgent.<init>(MultiDestinationSubscriberAgent.java:48)
mdc-subscriber-2_1  | 	at com.aeroncookbook.aeron.mdc.MultiDestinationSubscriber.main(MultiDestinationSubscriber.java:28)
mdc-publisher_1     | 12:38:00.405 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationPublisherAgent - launching aeron
mdc-publisher_1     | 12:38:00.415 [main] INFO com.aeroncookbook.aeron.mdc.MultiDestinationPublisherAgent - Media Driver directory is /dev/shm/aeron-root
mdc-publisher_1     | Exception in thread "main" java.util.NoSuchElementException
mdc-publisher_1     | 	at java.base/java.net.NetworkInterface$1.nextElement(NetworkInterface.java:415)
mdc-publisher_1     | 	at com.aeroncookbook.aeron.mdc.MultiDestinationPublisherAgent.localHost(MultiDestinationPublisherAgent.java:126)
mdc-publisher_1     | 	at com.aeroncookbook.aeron.mdc.MultiDestinationPublisherAgent.<init>(MultiDestinationPublisherAgent.java:44)
mdc-publisher_1     | 	at com.aeroncookbook.aeron.mdc.MultiDestinationPublisher.main(MultiDestinationPublisher.java:26)
```

**Comment by supergaga** [comment]
btw.   
https://aeroncookbook.com/aeron/multi-destination-cast/  

![9409B1ED-9120-4B5F-9F24-954FD15C4BA4](https://user-images.githubusercontent.com/13994894/122232163-9c502080-ceed-11eb-9340-7bcc3228a5d1.png)

**Comment by eleventy7** [comment]
Thanks for the correction - I've pushed a fix.

Re docker - `docker compose` can setup the network a bit incorrectly - for example, it tends to ignore the `ipv4_address`. Can you also try `docker-compose`? 

If no change with `docker-compose`, start the publisher, connect into running docker image using `docker exec -it aeron-mdc_mdc-publisher_1 bash` and share the output of `ip addr`?

**Comment by supergaga** [comment]
> Thanks for the correction - I've pushed a fix.
> 
> Re docker - `docker compose` can setup the network a bit incorrectly - for example, it tends to ignore the `ipv4_address`. Can you also try `docker-compose`?
> 
> If no change with `docker-compose`, start the publisher, connect into running docker image using `docker exec -it aeron-mdc_mdc-publisher_1 bash` and share the output of `ip addr`. The code is expecting a device named `eth0`.

Is this right？

```
➜  ~ docker exec -it aeron-mdc_mdc-publisher_1 bash
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
root@mdc-publisher:~/jar# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
3: ip6tnl0@NONE: <NOARP> mtu 1452 qdisc noop state DOWN group default qlen 1000
    link/tunnel6 :: brd ::
43: eth0@if44: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:0a:01:00:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.1.0.2/24 brd 10.1.0.255 scope global eth0
       valid_lft forever preferred_lft forever
root@mdc-publisher:~/jar#
```

**Comment by eleventy7** [comment]
Looks ok to me. I've pushed an update to the MDC sample - I had accidentally left a group size of 5 when preparing code examples for Tagged Flow Control. As part of this change I've expanded the scope of the exception handler, so it should just fallback to the host provided in docker-compose.yml. This should be reliable with `docker-compose` but may have issues with `docker compose`.

Note I have also updated to Aeron 1.34.0, so the docker image `aeroncookbook/jdk16:latest` will need to be rebuilt or pulled from docker again, as it includes the Aeron Agent jar for 1.34.0 only.
